Error when changing root user password in UI

Using the default JQM image, a root user is created by default.
When trying to change the password using the UI, an exception occurs in the backend WS.

```
09/06.15:33:13.269|ERROR|qtp909786389-48                                                       |                RestClientException|com.enioka.jqm.jdbc.DatabaseException: UPDATE RUSER SET LOGIN=?, LOCKED=?, EXPIRATION_DATE=?, LAST_MODIFIED=CURRENT_TIMESTAMP, EMAIL=?, FREETEXT=? WHERE ID=?
        at com.enioka.jqm.jdbc.DbConn.runUpdate(DbConn.java:131)
        at com.enioka.admin.MetaService.upsertUser(MetaService.java:1867)
        at com.enioka.jqm.ws.api.ServiceAdmin.setUser(ServiceAdmin.java:588)
        at com.enioka.jqm.ws.api.ServiceAdmin.setUser(ServiceAdmin.java:578)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:146)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:189)
        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$VoidOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:159)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:93)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:274)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:266)
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:253)
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:696)
        at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:397)
        at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:349)
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:358)
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:312)
        at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764)
        at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1665)
        at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:149)
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:202)
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1635)
        at com.enioka.jqm.ws.plumbing.LogFilter.doFilter(LogFilter.java:82)
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:210)
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1635)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:527)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:598)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1580)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1381)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1553)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1303)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
        at org.eclipse.jetty.server.Server.handle(Server.java:563)
        at org.eclipse.jetty.server.HttpChannel$RequestDispatchable.dispatch(HttpChannel.java:1598)
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:753)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:501)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:287)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
        at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:421)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:390)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:277)
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:199)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:411)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:969)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1194)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1149)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.sql.SQLIntegrityConstraintViolationException: integrity constraint violation: NOT NULL check constraint ; SYS_CT_10318 table: RUSER column: LOCKED
        at org.hsqldb/org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
        at org.hsqldb/org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
        at org.hsqldb/org.hsqldb.jdbc.JDBCPreparedStatement.fetchResult(Unknown Source)
        at org.hsqldb/org.hsqldb.jdbc.JDBCPreparedStatement.executeUpdate(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at tomcat.jdbc@10.1.20/org.apache.tomcat.jdbc.pool.StatementFacade$StatementProxy.invoke(StatementFacade.java:118)
        at jdk.proxy4/jdk.proxy4.$Proxy17.executeUpdate(Unknown Source)
        at com.enioka.jqm.jdbc.DbConn.runUpdate(DbConn.java:106)
        ... 67 common frames omitted
Caused by: org.hsqldb.HsqlException: integrity constraint violation: NOT NULL check constraint ; SYS_CT_10318 table: RUSER column: LOCKED
        at org.hsqldb/org.hsqldb.error.Error.error(Unknown Source)
        at org.hsqldb/org.hsqldb.Table.enforceRowConstraints(Unknown Source)
        at org.hsqldb/org.hsqldb.Table.generateAndCheckData(Unknown Source)
        at org.hsqldb/org.hsqldb.Table.insertSingleRow(Unknown Source)
        at org.hsqldb/org.hsqldb.StatementDML.update(Unknown Source)
        at org.hsqldb/org.hsqldb.StatementDML.executeUpdateStatement(Unknown Source)
        at org.hsqldb/org.hsqldb.StatementDML.getResult(Unknown Source)
        at org.hsqldb/org.hsqldb.StatementDMQL.execute(Unknown Source)
        at org.hsqldb/org.hsqldb.Session.executeCompiledStatement(Unknown Source)
        at org.hsqldb/org.hsqldb.Session.execute(Unknown Source)
        ... 74 common frames omitted
An unexpected error occurred while processing a REST call
09/06.15:33:13.271|INFO |qtp909786389-48                                                       |                RestClientException|a REST call failed Internal Server Error
09/06.15:33:13.271|INFO |qtp909786389-48                                                       |                    c.e.j.w.request|"PUT /ws/admin/user/68 HTTP/1.1" 500 - 5
```

--- a/jqm-all/jqm-ws/src/main/react/src/components/Users/UserAPI.ts
+++ b/jqm-all/jqm-ws/src/main/react/src/components/Users/UserAPI.ts
@@ -77,8 +77,9 @@ export const useUserAPI = () => {
     );
 
     const changePassword = useCallback(
-        (userId: string) => async (password: string) => {
-            return APIService.put(`${API_URL}/${userId}`, {
+        (user: User) => async (password: string) => {
+            return APIService.put(`${API_URL}/${user.id}`, {
+                ...user,
                 newPassword: password,
             })
                 .then(() => {

--- a/jqm-all/jqm-ws/src/main/react/src/components/Users/UsersPage.tsx
+++ b/jqm-all/jqm-ws/src/main/react/src/components/Users/UsersPage.tsx
@@ -41,7 +41,7 @@ const UsersPage: React.FC = () => {
     const [expirationDate, setExpirationDate] = useState<Date | null>(null);
     const [userRoles, setUserRoles] = useState<number[] | null>(null);
     const [changePasswordUserId, setChangePasswordUserId] = useState<
-        string | null
+        number | null
     >(null);
     const {
         users,
@@ -316,7 +316,7 @@ const UsersPage: React.FC = () => {
                 {changePasswordUserId !== null && (
                     <ChangePasswordDialog
                         closeDialog={() => setChangePasswordUserId(null)}
-                        changePassword={changePassword(changePasswordUserId)}
+                        changePassword={changePassword(users?.find(user => user.id === changePasswordUserId)!)}
                     />
                 )}
                 {showCreateDialog && (

